### ISSUE TITLE
Emtpy kw only dataclass has incorrect type hint for __init__ signature

### ISSUE THREAD

**Issue by LunarLanding** [microsoft/pylance-release#3916]
## Environment data

-   Language Server version: 2023.2.10
-   OS and version: OSX 13.1
-   Python version (& distribution if applicable, e.g. Anaconda): miniconda python 3.10.8

## Code Snippet

```python
from dataclasses import dataclass
@dataclass(kw_only=True)
class A:
    pass
A() # shows (*args:Any,**kwargs:Any)
```
Screenshot
<img width="309" alt="image" src="https://user-images.githubusercontent.com/4441338/217210638-1a6e602c-57d5-4c54-815d-fb5fb179743f.png">

## Repro Steps

1. Enter code above and show the type hint

## Expected behavior

Typehint shows no args or kwargs, empty signature.

## Actual behavior

Type hint shows any args or kwargs accepted.

## Logs

Collected in empty vscode instance, by turning on trace, clearing, restarting, pasting code into unnamed file, and hovering over last line.

```
[Info  - 10:56:22] (13001) Pylance language server 2023.2.10 (pyright 01758105) starting
[Info  - 10:56:22] (13001) Server root directory: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist
[Info  - 10:56:22] (13001) Starting service instance "<default>"
[Info  - 10:56:22] (13001) No pyproject.toml file found.
[Info  - 10:56:22] (13001) Setting pythonPath for service "<default>": "$CONDA_PREFIX/bin/python"
[Warn  - 10:56:22] (13001) stubPath typings is not a valid directory.
[Info  - 10:56:22] (13001) Assuming Python version 3.10
[Info  - 10:56:22] (13001) Assuming Python platform Darwin
[Info  - 10:56:22] (13001) Search paths for <default>
[Info  - 10:56:22] (13001)   $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib
[Info  - 10:56:22] (13001)   .
[Info  - 10:56:22] (13001)   typings
[Info  - 10:56:22] (13001)   $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stubs/...
[Info  - 10:56:22] (13001)   $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/bundled/stubs
[Info  - 10:56:22] (13001)   $CONDA_PREFIX/lib/python3.10
[Info  - 10:56:22] (13001)   $CONDA_PREFIX/lib/python3.10/lib-dynload
[Info  - 10:56:22] (13001)   $HOME/.local/lib/python3.10/site-packages
[Info  - 10:56:22] (13001)   $CONDA_PREFIX/lib/python3.10/site-packages
[Info  - 10:56:22] (13001) Searching for source files
[Info  - 10:56:22] (13001) No source files found.
(13001) [IDX(FG)] index libraries  (index) ...
(13001) [IDX(FG)]   read stdlib indices (16ms)
(13001) [IDX(FG)] index libraries  (index) [succeed] (17ms)
[Info  - 10:56:22] (13001) No pyproject.toml file found.
[Info  - 10:56:22] (13001) Setting pythonPath for service "<default>": "$CONDA_PREFIX/bin/python"
[Warn  - 10:56:22] (13001) stubPath typings is not a valid directory.
[Info  - 10:56:22] (13001) Assuming Python version 3.10
[Info  - 10:56:22] (13001) Assuming Python platform Darwin
[Info  - 10:56:22] (13001) Search paths for <default>
[Info  - 10:56:22] (13001)   $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib
[Info  - 10:56:22] (13001)   .
[Info  - 10:56:22] (13001)   typings
[Info  - 10:56:22] (13001)   $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stubs/...
[Info  - 10:56:22] (13001)   $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/bundled/stubs
[Info  - 10:56:22] (13001)   $CONDA_PREFIX/lib/python3.10
[Info  - 10:56:22] (13001)   $CONDA_PREFIX/lib/python3.10/lib-dynload
[Info  - 10:56:22] (13001)   $HOME/.local/lib/python3.10/site-packages
[Info  - 10:56:22] (13001)   $CONDA_PREFIX/lib/python3.10/site-packages
[Info  - 10:56:22] (13001) Searching for source files
[Info  - 10:56:22] (13001) No source files found.
(13001) [IDX(FG)] index libraries  (index) ...
(13001) [IDX(FG)]   read stdlib indices (0ms)
(13001) [IDX(FG)] index libraries  (index) [succeed] (1ms)
[Info  - 10:56:22] (13001) Background analysis(1) root directory: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist
[Info  - 10:56:22] (13001) Background analysis(1) started
(13001) Background analysis message: setConfigOptions
(13001) Background analysis message: setImportResolver
(13001) Background analysis message: ensurePartialStubPackages
(13001) [FG] parsing: Untitled-1 (16ms)
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (83ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/builtins.pyi (15ms)
(13001) [FG] binding: Untitled-1 (0ms)
(13001) Background analysis message: setTrackedFiles
(13001) Background analysis message: markAllFilesDirty
(13001) Background analysis message: setFileOpened
(13001) Background analysis message: setConfigOptions
(13001) Background analysis message: setImportResolver
(13001) Background analysis message: ensurePartialStubPackages
(13001) Background analysis message: setTrackedFiles
(13001) Background analysis message: markAllFilesDirty
(13001) Background analysis message: getSemanticTokens full
(13001) [BG(1)] getSemanticTokens full at Untitled-1 ...
(13001) [BG(1)]   parsing: Untitled-1 (17ms)
(13001) [BG(1)]   parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (84ms)
(13001) [BG(1)]   binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/builtins.pyi (17ms)
(13001) [BG(1)]   binding: Untitled-1 (0ms)
(13001) [BG(1)] getSemanticTokens full at Untitled-1 (120ms)
(13001) Background analysis message: analyze
(13001) [BG(1)] analyzing: Untitled-1 ...
(13001) [BG(1)]   checking: Untitled-1 (3ms)
(13001) [BG(1)] analyzing: Untitled-1 (3ms)
(13001) Background analysis message: resumeAnalysis
(13001) Background analysis message: setFileOpened
(13001) Background analysis message: markFilesDirty
(13001) [FG] parsing: Untitled-1 (10ms)
(13001) [FG] binding: Untitled-1 (0ms)
(13001) Background analysis message: analyze
(13001) [BG(1)] analyzing: Untitled-1 ...
(13001) [BG(1)]   parsing: Untitled-1 (9ms)
(13001) [BG(1)]   binding: Untitled-1 (0ms)
(13001) [BG(1)]   checking: Untitled-1 ...
(13001) [BG(1)]     parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing.pyi [fs read 2ms] (39ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing.pyi (3ms)
(13001) [BG(1)]     parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 0ms] (3ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing_extensions.pyi (1ms)
(13001) [BG(1)]     parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 0ms] (24ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (1ms)
(13001) [BG(1)]     parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/types.pyi [fs read 1ms] (10ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/types.pyi (1ms)
(13001) [BG(1)]     parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/abc.pyi [fs read 1ms] (1ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/abc.pyi (1ms)
(13001) [BG(1)]     parsing: $CONDA_PREFIX/lib/python3.10/dataclasses.py [fs read 1ms] (34ms)
(13001) [BG(1)]     binding: $CONDA_PREFIX/lib/python3.10/dataclasses.py (4ms)
(13001) [BG(1)]     parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/dataclasses.pyi [fs read 0ms] (6ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/dataclasses.pyi (1ms)
(13001) [BG(1)]     parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 0ms] (0ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/collections/abc.pyi ...
(13001) [BG(1)]       parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 0ms] (1ms)
(13001) [BG(1)]       binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_collections_abc.pyi (0ms)
(13001) [BG(1)]     binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/collections/abc.pyi (1ms)
(13001) [BG(1)]   checking: Untitled-1 (160ms)
(13001) [BG(1)] analyzing: Untitled-1 (169ms)
(13001) Background analysis message: getDiagnosticsForRange
(13001) Background analysis message: getSemanticTokens delta
(13001) [BG(1)] getSemanticTokens delta previousResultId:1675763782447 at Untitled-1 (1ms)
(13001) Background analysis message: resumeAnalysis
(13001) Background analysis message: getDiagnosticsForRange
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing.pyi [fs read 1ms] (43ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing.pyi (3ms)
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 1ms] (3ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/typing_extensions.pyi (2ms)
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 0ms] (25ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (1ms)
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/types.pyi [fs read 0ms] (8ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/types.pyi (2ms)
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/abc.pyi [fs read 1ms] (1ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/abc.pyi (0ms)
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/dataclasses.pyi [fs read 0ms] (5ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/dataclasses.pyi (1ms)
(13001) [FG] parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 0ms] (0ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/collections/abc.pyi ...
(13001) [FG]   parsing: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 0ms] (0ms)
(13001) [FG]   binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/_collections_abc.pyi (0ms)
(13001) [FG] binding: $HOME/.vscode/extensions/ms-python.vscode-pylance-2023.2.10/dist/typeshed-fallback/stdlib/collections/abc.pyi (1ms)

```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in an upcoming version.

**Comment by bschnurr** [comment]
This issue has been fixed in prerelease version 2023.2.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023221-8-February-2023-PreRelease)
